Add async data helper function (#34)
infinum/ngx-nuts-and-bolts: libs/ngx-nuts-and-bolts/src/lib/testing/async-data/README.md

# Async data helper

> If and when returning mocked data from testing service user should avoid using something like:

Reviewer: ```suggestion
If and when returning mocked data from the testing service, users should avoid doing something like this:
```

```ts
public getData(data: TData): Observable<TData>{
	return of(data);
}
```

> Above example creates observable from the data but it returns it synchronously which isn't how the services that make API calls behave in real world applications.

Reviewer: ```suggestion
The example above creates an observable from the data but it [returns it synchronously](https://stackblitz.com/edit/rxjs-hx8yos?devtoolsheight=60). This isn't how services that make API calls behave in real world applications.
```

> The purpose of creating test doubles for services is to mock real world services behavior as much as possible, so with that in mind it is good idea to fake the async behavior of such services, which can be achieved in multiple ways.

Reviewer: ```suggestion
The purpose of creating test doubles for services is to mock the behavior of real world services as closely as possible. With that in mind, it is a good idea to fake the async behavior of such services. This can can be achieved in multiple ways.
```

> This specific implementation of helper uses `observeOn` operator which applies `asyncScheduler` which creates new macrotask and puts it next in event loop queue.

Reviewer: ```suggestion
This specific implementation uses `observeOn` operator with `asyncScheduler`, creating a new macrotask that places the observable in the event loop queue.
```

## 1. Usage

> In service test double just call `asyncData` function with the mocked data you want to return asynchronously.

Reviewer: ```suggestion
When returning the mock data in test doubles, simply wrap the data using `asyncData` function and the data will be returned asynchronously.
``

# Async data helper

If and when returning mocked data from the testing service, users should avoid doing something like this:

```ts
public getData(data: TData): Observable<TData>{
	return of(data);
}
```

Above example creates observable from the data but it returns it synchronously which isn't how the services that make API calls behave in real world applications.
The purpose of creating test doubles for services is to mock the behavior of real world services as closely as possible. With that in mind, it is a good idea to fake the async behavior of such services. This can can be achieved in multiple ways.
This specific implementation of helper uses `observeOn` operator which applies `asyncScheduler` which creates new macrotask and puts it next in event loop queue.

## 1. Usage

When returning the mock data in test doubles, simply wrap the data using `asyncData` function and the data will be returned asynchronously.

> ``

Reviewer: Remove
